presto-main/src/test/java/io/prestosql/execution/TestDynamicFiltersCollector.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.execution.DynamicFiltersCollector.VersionedDynamicFilterDomains;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.sql.planner.plan.DynamicFilterId;
+import org.testng.annotations.Test;
+
+import static io.prestosql.execution.DynamicFiltersCollector.INITIAL_DYNAMIC_FILTERS_VERSION;
+import static io.prestosql.spi.predicate.Domain.multipleValues;
+import static io.prestosql.spi.predicate.Domain.singleValue;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestDynamicFiltersCollector
+{
+    @Test
+    public void testDynamicFiltersCollector()
+    {
+        DynamicFilterId filter = new DynamicFilterId("filter");
+        DynamicFiltersCollector collector = new DynamicFiltersCollector(() -> {});
+
+        VersionedDynamicFilterDomains domains = collector.getDynamicFilterDomains(INITIAL_DYNAMIC_FILTERS_VERSION);
+
+        // there should be no domains initially
+        assertEquals(domains.getVersion(), INITIAL_DYNAMIC_FILTERS_VERSION);
+        assertEquals(domains.getDynamicFilterDomains(), ImmutableMap.of());
+
+        Domain initialDomain = multipleValues(BIGINT, ImmutableList.of(1L, 2L, 3L));
+        collector.collectDynamicFilterDomains(ImmutableMap.of(filter, initialDomain));
+
+        domains = collector.getDynamicFilterDomains(INITIAL_DYNAMIC_FILTERS_VERSION);
+        assertEquals(domains.getVersion(), 1L);
+        assertEquals(domains.getDynamicFilterDomains(), ImmutableMap.of(filter, initialDomain));
+
+        // make sure domains are still available when requested again with old version
+        domains = collector.getDynamicFilterDomains(INITIAL_DYNAMIC_FILTERS_VERSION);
+        assertEquals(domains.getVersion(), 1L);
+        assertEquals(domains.getDynamicFilterDomains(), ImmutableMap.of(filter, initialDomain));
+
+        // make sure domains are intersected
+        collector.collectDynamicFilterDomains(ImmutableMap.of(filter, multipleValues(BIGINT, ImmutableList.of(2L))));
+        collector.collectDynamicFilterDomains(ImmutableMap.of(filter, multipleValues(BIGINT, ImmutableList.of(3L))));
```

Comment:
This one of these calls should have a new value in addition to  an existing value.  With this code, the implementation could simply ignore the updates and the test would still work.